pkg/apis/networking/types.go

```diff
@@ -544,3 +544,68 @@ type ServiceBackendPort struct {
 	// +optional
 	Number int32
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ClusterCIDRConfig is the Schema for the clustercidrconfigs API.
+type ClusterCIDRConfig struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+
+	Spec   ClusterCIDRConfigSpec
+	Status ClusterCIDRConfigStatus
+}
+
+// ClusterCIDRConfigSpec defines the desired state of ClusterCIDRConfig.
+type ClusterCIDRConfigSpec struct {
+	// NodeSelector defines which nodes the config is applicable to.
+	// An empty or nil NodeSelector functions as a default that applies to all nodes.
+	// This field is immutable.
+	// +optional
+	NodeSelector *api.NodeSelector
+
+	// IPv4 defines the IPv4 CIDR and the PerNodeMaskSize.
+	// At least one of IPv4 or IPv6 must be provided. If both are
+	// provided, the number of IPs allocated to each must be the same
+	// (32 - ipv4.perNodeMaskSize).
+	// This field is immutable.
+	// +optional
+	IPv4 *CIDRConfig
+
+	// IPv6 defines the IPv4 CIDR and the PerNodeMaskSize.
+	// At least one of IPv4 or IPv6 must be provided. If both are
+	// provided, the number of IPs allocated to each must be the same
+	// (128 - ipv6.perNodeMaskSize).
+	// This field is immutable.
+	// +optional
+	IPv6 *CIDRConfig
```

Comment:
Are these configs mutually exclusive with current cli arguments? current cli argument are defaulted. Shouldn't we do the same here?

Reply:
With the updated API this field is obsolete, please refer the API PR: https://github.com/kubernetes/kubernetes/pull/111123